### ISSUE TITLE
Please keep in mind that many users will want to submit their LaTeX code to the arXiv

### ISSUE THREAD

**Issue by marmotghost** [learnlatex/learnlatex.github.io#97]
Arguably LaTeX is very widespread in universities and research institutions. Many authors wish to submit their article eventually to the arXiv. Therefore, it could make sense to add comments when discussing `lualatex`, `xelatex`, `biblatex` and unicode characters. Please inform the users these are either not at all or only partially supported by the arXiv. To give an example, I find that the current comparison between `BibTeX` and `biblatex` misses a warning concerning `biblatex`, see [here](https://github.com/plk/biblatex/wiki/biblatex-and-the-arXiv) for more information. AFAIK know, neither `lualatex` nor `xelatex` are currently supported. Unicode characters are not supported in the title and so on fields and were not allowed in the `.tex` files, whether they are allowed by now I do not know.

**Comment by davidcarlisle** [comment]
@marmotghost  I have some sympathy with this but if we are not careful documenting the contortions required for arxiv will be the hardest thing taught on the course (harder than the actual basic latex features covered).

For the overleaf case, we could (somewhere) suggest people use the option to use texlive2016 (I think it is 2016 needed)?  I don't have space at latexcgi.xyz to run two texlives so I can not offer that option there currently, but as latexcgi doesn't provide any way to save the tex source it is unlikely to be used for actual document preparation (I hope) so perhaps that is not a problem.

While not specifically mentioning arxiv lesson12 does say

> Having been around for much longer than biblatex, the BibTeX workflow is more established than biblatex, meaning that many publishers and journals expect bibliographies generated via the BibTeX workflow. Those publishers cannot or generally do not accept submissions using biblatex

Which is essentially covering the same ground.

**Comment by josephwright** [comment]
I think it's also worth remembering that _brand new_ users are less likely to be PIs, and it's the latter who will be/should be in charge of sending results to arXiv or anywhere else 'external'.

**Comment by marmotghost** [comment]
@josephwright Not really. There are more than enough examples in which younger researchers went ahead and added all sort of fancy stuff from the internet only to get frustrated at the submission because, say, arXiv switched to the TeX (rather than LaTeX) mode because it found some unicode characters, or some Feynman diagrams caused problems because they required Lualatex. The PI does not always care about the details because they just did not try out any of these so they did not know the problems, just avoided them because they never used anything but PDFLaTeX. So I do think that it cannot hurt to mention these pitfalls. Sometimes the delay due to submission issues can cost the priority on a result. (And there are examples in which PhD students wanted to upload their thesis in which they wanted to do things in a "more modern way".) The arXiv administrators can also get quite angry if one uploads junk that they do not like.

**Comment by marmotghost** [comment]
@davidcarlisle This goes certainly in the right direction but I'd say that mentioning the arXiv is more important than mentioning the publishers. Many journals support arXiv upload, i.e. once you have it on the arXiv, you are done.

**Comment by josephwright** [comment]
@marmotghost The wider point is that we can't in 16 short lessons get someone ready to write a paper, thesis, etc.: we can get them ready to learn what they need from other sources. We also have to balance different user needs: for every person in physics (etc.) who might use arXiv, there is likely a humanities user who wants multiple character sets, etc. That's also the case with 'publishers': ones in physics might take arXiv data, but in other areas they won't, but will provide BibTeX styles. 

I'm wary of picking a 'lowest common denominator' set of restrictions which rule out any modern-ish developments. (`biblatex` for example is 10+ years old.)

**Comment by marmotghost** [comment]
@josephwright I understand this but still think that one could, or even should, put a warning somewhere in order not to unnecessarily upset people. At least not if the idea is that this resource is better than the various introductions to LaTeX that already exist. It is kind of frustrating to run into such problems only to read then on some Q & A sites that it is a well known fact that arXiv does not support lualatex. You could argue that the arXiv should make this clear, the problem with the argument is that user look at their documentation (if at all) when submitting and not when starting to prepare a document.

**Comment by josephwright** [comment]
We could put in some generalised 'take care' info, for example

> If you are planning to send your LaTeX _sources_ to public destinations, such as publishers, conference organisers 
> or pre-print servers (e.g. arXiv), you should check what restrictions they impose. In particular, the availability of Unicode
> engines and up-to-date packages may not be guaranteed.

**Comment by marmotghost** [comment]
@josephwright Almost perfect! I'd add just some clarifications.

> 
If you are planning to send your LaTeX sources to destinations which process them, such as publishers, conference organisers or pre-print servers (e.g. arXiv), you should check what restrictions they impose. In particular, the availability of Unicode engines and/or engines that call external tools (such as lualatex and xelatex) and up-to-date packages may not be guaranteed.

**Comment by josephwright** [comment]
OK, the question is where the proviso goes: in the Unicode lesson?

**Comment by marmotghost** [comment]
@josephwright Good question. I would have expected to see a section on "Why and when should one use LaTeX?" in which the user could learn whether it makes sense to use LaTeX, and for which purposes. In such a section one could mention the limitations and pitfalls. Alternatively one could add a section on "Limitations and pitfalls" in which one could mention these things. But I do think one should mention them prominently, it is all-too-frustrating if you learn certain tricks with `lualatex`, say, just to find out that you can't really use them for your publication.

**Comment by josephwright** [comment]
@marmotghost I think if possible I'd like to keep mentions of alternative engines, etc. 'inside' their parts of the site. Would for example it be OK to mention this in the Unicode lesson, with a cross-ref or similar from the `biblatex` subsection of the bibliography one?

**Comment by marmotghost** [comment]
@josephwright Sure. That would be great. My only concern was that users may add a lot of things only to learn that they could not use it for their paper. Sometimes the advantages that new tools bring are not essential, for instance may order the references in the order they cite them, so they do not really care whether the tool they use is capable of sorting umlauts, say. Similarly it might be worthwhile to typeset an `é` as `\'e` if you have only a few occurrences in your text because then you may be able to use `pdflatex`, which is faster, and, more importantly, submit the opus to the arXiv.

**Comment by josephwright** [comment]
I've added the text but decided it might be best earlier: let's see if people like it where I've put it.

**Comment by josephwright** [comment]
@marmotghost Did you see https://blog.arxiv.org/2020/09/24/tex-live-2020-release-oct-1-2020/? While it's of course right to add general advice about publishers, etc., this does make a difference for at least some parts of the arXiv situation.

**Comment by marmotghost** [comment]
@josephwright Thanks! No, I did not read this before. Yet, unless I miss something, still lualatex etc. seem not to be supported. If that's the case, I think it is important to warn the users appropriately.
